src/Kernel/Object.class.st

```diff
@@ -824,6 +824,7 @@ Object >> doesNotUnderstand: aMessage [
 	"Testing: (3 activeProcess)"
 		
 	| exception resumeValue |
+	JS ifConnected: [JS debugger].
```

Comment:
Why in case of JS we should forbid MessageNotUnderstood exceptions?
It will break any code like "[...] on: MessageNotUnderstood do: [...]" even if it is not a good practice.

Reply:
To be able inject native JS debugger you should implement a debugger.
Check debugger API by "Smalltalk tools debugger" senders

Reply:
This is not forbidding MessageNotUnderstood exceptions, only adding a JS-level halt as a side-effect along the way (e.g., something that will make the Chrome DevTools stop, in a resumable way). "<object>.debugger" is a common JS idiom; another would be to throw a particular JS exception.

Reply:
But #doesNotUnderstood:  is not the place where debugger is opening. 
MessageNotUnderstood exception is not different than any other error like NotFound for example.

Why JS.debugger can't be called during UnhandledException processing? And it was my proposal to implement own debugger and register it in the tools. It will just call JS level halt.

Reply:
Ah, okay. Those both make sense.